app.controllers.js

```diff
@@ -21,22 +20,37 @@ const getArticleById = (req, res, next) => {
     })
     .then((article) => {
         res.status(200).send({ article });
-    })
-    .catch((err) => {
-        res.status(err.status || 500).send({msg: err.msg})
-})
+    }).catch(err => {
+        next(err);
+    });
 }
 
-const getCommentsByArticleId = (req, res) => {
+const getCommentsByArticleId = (req, res, next) => {
     const {article_id} = req.params;
+
+    fetchArticleById(article_id).catch(next);
```

Comment:
Instead of doing 2 separate promise chains, I would probably try to make it into one. This will still work, but is definitely an unusual way of doing this!